### PR TITLE
Build Faucet images with GitHub Actions

### PR DESCRIPTION
**Pull request by kchulj**

### CODE REVIEW

File: .github/workflows/faucet-docker.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        run: |
+          set -xeu
+          set -o pipefail
+          if [[ $github_ref == /refs/tags/* ]]
+          then
+            FAUCET_VERSION="${github_ref#"/refs/tags/"}-${github_sha}"
+          else
+            FAUCET_VERSION="$github_sha"
+          fi
+          echo "FAUCET_VERSION=$FAUCET_VERSION" >> $GITHUB_ENV
```
Comment (Geod24) [inline .github/workflows/faucet-docker.yml:42]: What's the point of this ?

Comment (kchulj) [reply .github/workflows/faucet-docker.yml:42]: I put it because agora has this step.

**Comment by Geod24** [comment]
Just disabled docker hub

File: dub.selections.json [hunk 2]
```diff
@@ -2,26 +2,25 @@
...
+		"bitblob": {"path":"submodules/bitblob/"},
+		"botan": {"path":"submodules/botan/"},
+		"d2sqlite3": {"path":"submodules/d2sqlite3/"},
+		"diet-ng": {"path":"submodules/diet-ng/"},
+		"dyaml": {"path":"submodules/dyaml/"},
+		"eventcore": {"path":"submodules/eventcore/"},
+		"libasync": {"path":"submodules/libasync/"},
+		"libsodiumd": {"path":"submodules/libsodiumd/"},
+		"localrest": {"path":"submodules/localrest/"},
+		"mir-core": "0.2.2",
```
Comment (Geod24) [inline dub.selections.json:16]: This needs to be removed before this PR can be ready. What pulls it in ?

Comment (kchulj) [reply dub.selections.json:16]: I haven't pushed the changes yet. Will do clean-up and rebase before I mark this as ready.

**Comment by kchulj** [comment]
This is blocking the GitHub action. Any ideas?
```
No package file found in /root/faucet/submodules/localrest/submodules/vibe-core/, expected one of dub.json/dub.sdl/package.json
```

File: dub.selections.json [hunk 3]
```diff
@@ -2,26 +2,24 @@
...
-		"taggedalgebraic": "0.11.19",
-		"tinyendian": "0.2.0",
-		"vibe-core": "1.13.0",
-		"vibe-d": "0.9.3-rc.1"
+		"openssl": {"path":"submodules/openssl/"},
+		"stdx-allocator": {"path":"submodules/stdx-allocator/"},
+		"taggedalgebraic": {"path":"submodules/taggedalgebraic/"},
+		"tinyendian": {"path":"submodules/tinyendian"},
+		"vibe-core": {"path":"submodules/vibe-core"},
+		"vibe-d": {"path":"submodules/vibe-core"}
```
Comment (Geod24) [inline dub.selections.json:23]:
```suggestion
		"vibe-d": {"path":"submodules/vibe-d"}
```

**Comment by kchulj** [comment]
Green.
I will do a quick check before marking as ready.

**Pull request merged**